jep/1/README.adoc

```diff
@@ -0,0 +1,767 @@
+However, long open-ended discussions are not recommended on mailing lists.
+Strategies to keep the discussion efficient include:
+
+* setting up a series of in-person, or video-conferencing sessions to
+  discuss the JEP with necessary stakeholders.
+* having the JEP author accept private comments in the early design phases
+* setting up a wiki page, etc.
+* committing and reviewing small concrete changes via Pull Requests
+  rather than large sweeping changes
+
+JEP authors should use their discretion here.
+
+The JEP author may also ask JEP editors for further feedback regarding the
+style and consistency of a JEP and its readiness for review by the BDFL.
+
+As updates are necessary, the JEP author and other contributors
+should push commits to their fork of the JEP repository footnoteref:[repo],
+and submit pull requests targeting the JEP's feature branch.
+
+[[review]]
+==== JEP Review
+
+Once the author believes a JEP is complete,
+they request the BDFL review the JEP for acceptance, usually via
+an email to the jenkinsci-dev@ mailing list.
+For a JEP that is predetermined to be acceptable
+(e.g., it is an obvious win as-is and/or its implementation has already been checked in)
+the BDFL may also initiate a JEP review, first notifying the JEP author and
+giving them a chance to make revisions beforehand.
+The BDFL and their chosen consultants then review the JEP.
+They will resolve the JEP as "Accepted" or "Rejected",
+or keep it as "Draft" sending it back to the JEP author for revision.
+
+The final authority for JEP resolution is the BDFL. However, whenever a new
+JEP is put forward, any core developer that believes they are suitably
+experienced to make the final decision on that JEP may offer to serve as
+the BDFL's delegate (or "JEP czar") for that JEP. If their self-nomination
+is accepted by the other core contributors and the BDFL, then they will have
+the authority to accept (or reject) that JEP. This process happens most
+frequently with JEPs where the BDFL has agreed in principle that
+*something* needs to be done, but there are details that need to be worked out
+before the JEP can be accepted.
+
+If the final decision on a JEP is to be made by a delegate rather than
+directly by the BDFL, this will be recorded by including the
+"BDFL-Delegate" header in the JEP.
+
+JEP review and resolution may also occur on a list other than jenkinsci-dev@ In
+this case, the "Discussions-To" heading in the JEP will identify the
+appropriate alternative list where discussion, review and pronouncement on the
+JEP will occur.
+
+[[accepted]]
+==== Accepting a JEP
+
+For a JEP to be "Accepted" it must meet certain minimum criteria:
+
+* It must be a clear and complete description of the proposed enhancement.
+* The enhancement must represent a net improvement.
+* The proposed implementation, if applicable, must be solid, must not
+  complicate Jenkins unduly, and must be the same license as the component the
```

Comment:
I am reposting my comment about "same license" in https://github.com/jenkinsci/jep/pull/8#discussion_r139896232, because IMHO it worth discussion:

> "same license" is maybe too strong. But it's better than legal holywars about "compatible licenses".
>
> OTOH some exceptions could be useful. E.g. some companies strictly require "Apache v.2", because MIT is not explicit about the patent release. By setting this condition we explicitly require companies to do reference implementation with the MIT license, and hence such potential contributors may have no way to comply with the process without getting a legal-approved exception (it hurts :( ).
> 
> I would just pre-approve MIT, Apache and BSD licenses for reference implementations, independently of the component license. GPL/LGPL is more complicated in the case of Jenkins.

@daniel-beck @rtyler

Reply:
I don't even want to think about the mess that mixed-licensed (i.e. half one license, half the other; not dual licensed!) components and correctly working with that would be. IANAL after all.

Reply:
@oleg-nenashev I would assume most JEPs will refer to core or many of the "essential" plugins, all of which are MIT licensed, such that anybody contributing to those repositories is expected to contribute under the existing (MIT) license.

I believe we should keep this text as-is until we have a reason to change it at some later date.

Reply:
@oleg-nenashev Adding to "Reasoning".  Resolved?

Reply:
RESOLVED